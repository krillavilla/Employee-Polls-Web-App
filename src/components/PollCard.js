import { Link } from 'react-router-dom';
import UserAvatar from './UserAvatar';

export default function PollCard({ question, author }) {
  if (!question || !author) return null;
  
  return (
    <div style={{ 
      border: '1px solid #ddd', 
      padding: 16, 
      borderRadius: 8, 
      marginBottom: 12,
      backgroundColor: 'white',
      boxShadow: '0 2px 4px rgba(0,0,0,0.1)'
    }}>
      <div style={{ display: 'flex', alignItems: 'center', gap: 8, marginBottom: 12 }}>
        <UserAvatar src={author.avatarURL} alt={author.name} />
        <strong>{author.name} asks:</strong>
      </div>
      <h4 style={{ margin: '8px 0', color: '#333' }}>Would you rather</h4>
      <p style={{ color: '#666', marginBottom: 16 }}>
        ... {question.optionOne.text} ... or ... {question.optionTwo.text} ...
      </p>
      <Link 
        to={`/questions/${question.id}`}
        style={{
          color: '#007bff',
          textDecoration: 'none',
          fontWeight: 'bold'
        }}
      >
        View Poll
      </Link>
    </div>
  );
}