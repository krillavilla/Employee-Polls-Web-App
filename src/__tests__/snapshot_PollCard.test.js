import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import PollCard from '../components/PollCard';

it('PollCard matches snapshot', () => {
  const question = {
    id: 'q1',
    author: 'sarahedo',
    timestamp: 1467166872634,
    optionOne: { text: 'Build with JavaScript', votes: [] },
    optionTwo: { text: 'Build with TypeScript', votes: [] }
  };
  
  const author = { 
    id: 'sarahedo', 
    name: 'Sarah Edo', 
    avatarURL: 'https://example.com/avatar.jpg' 
  };

  const { container } = render(
    <MemoryRouter>
      <PollCard question={question} author={author} />
    </MemoryRouter>
  );
    
  expect(container.firstChild).toMatchSnapshot();
});
