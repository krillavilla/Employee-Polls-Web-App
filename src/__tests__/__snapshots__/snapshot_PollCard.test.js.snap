// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PollCard matches snapshot 1`] = `
<div
  style="border: 1px solid #ddd; padding: 16px; border-radius: 8px; margin-bottom: 12px; background-color: white; box-shadow: 0 2px 4px rgba(0,0,0,0.1);"
>
  <div
    style="display: flex; align-items: center; gap: 8px; margin-bottom: 12px;"
  >
    <img
      alt="Sarah Edo"
      height="32"
      src="https://example.com/avatar.jpg"
      style="border-radius: 50%;"
      width="32"
    />
    <strong>
      Sarah Edo
       asks:
    </strong>
  </div>
  <h4
    style="margin: 8px 0px; color: rgb(51, 51, 51);"
  >
    Would you rather
  </h4>
  <p
    style="color: rgb(102, 102, 102); margin-bottom: 16px;"
  >
    ... 
    Build with JavaScript
     ... or ... 
    Build with TypeScript
     ...
  </p>
  <a
    href="/questions/q1"
    style="color: rgb(0, 123, 255); text-decoration: none; font-weight: bold;"
  >
    View Poll
  </a>
</div>
`;
